app/src/main/java/net/aiscope/gdd_app/ui/capture/CaptureImagePresenter.kt

```diff
@@ -6,20 +6,26 @@ import timber.log.Timber
 
 class CaptureImagePresenter(
     val view: CaptureImageView,
-    val repository: SampleRepository
+    val repository: SampleRepository,
+    var receivedImageCapture: Boolean = false
 ) {
 
     fun handleCaptureImageButton(imageName: String) {
-        Timber.tag("Taking Photo").d("button pressed")
-        view.takePhoto(imageName) { file ->
-            Timber.tag("Taking Photo").d(file?.absolutePath)
-            if (file == null) {
-                view.notifyImageCouldNotBeTaken()
-            } else {
-                val sample = repository.current().addNewlyCapturedImage(file)
-                repository.store(sample)
+        if (receivedImageCapture){
+            Timber.tag("Taking Photo").d("Already processing photo")
+        } else {
+            receivedImageCapture = true
+            Timber.tag("Taking Photo").d("button pressed")
+            view.takePhoto(imageName) { file ->
+                Timber.tag("Taking Photo").d(file?.absolutePath)
```

Comment:
Shouldn't `receivedImageCapture` be set to false somewhere here? Especially if there was an error, we want to let the user to be able to try again.

Reply:
Good point. Not quite sure to do it in or outside the `suspend onPhotoReceived` though.  I mean, ideally we wait for that to complete. It's also possible we never make it to that, either cause the `takePhoto` does not succeed here: https://github.com/theaiscope/GDD-app/blob/cbfecac1e7ed1e896c2e2e0f48dd725b1ff9a603/app/src/main/java/net/aiscope/gdd_app/ui/capture/CaptureImageActivity.kt#L97

or if something goes wrong in the `onPhotoReceived` itself